verification/src/solidity/compiler_fetcher.rs

```diff
@@ -142,8 +228,8 @@ fn create_executable(path: &Path) -> Result<File, std::io::Error> {
 impl Fetcher for CompilerFetcher {
     type Error = FetchError;
     async fn fetch(&self, ver: &CompilerVersion) -> Result<PathBuf, Self::Error> {
-        let compiler_info = self
-            .releases
+        let compiler_versions = self.compiler_versions.versions.read().await;
```

Comment:
```rust
let compiler_info = {
  let compiler_versions = self.compiler_versions.versions.read().await;
  compiler_versions
            .0
            .get(ver)
            .ok_or_else(|| FetchError::NotFound(ver.clone()))?.clone()
}
```
this will heavilly reduce lock usage

Reply:
Oh, didn't notice, sure